cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -101,16 +101,20 @@ public void createOrUpdate(Reconciliation reconciliation, ConfigMap assemblyCm,
         } catch (Exception e) {
             return Future.failedFuture(e);
         }
+
         Service service = kafka.generateService();
         Service headlessService = kafka.generateHeadlessService();
-        ConfigMap metricsConfigMap = kafka.generateMetricsConfigMap();
+        ConfigMap logAndMetricsConfigMap = kafka.generateMetricsAndLogConfigMap(
+                kafka.getLogging() instanceof ExternalLogging ?
+                        configMapOperations.get(namespace, ((ExternalLogging) kafka.getLogging()).name) :
```

Comment:
There's a problem with this: the `get()` will block. I don't think you should fix that in this PR. I think you should wait until @ppatierno has merged his work, which will alter this chain of `compose()` and allow passing state between then. With that in place you should be able to add this getting of the configmap into the `compose()` chain.

Reply:
Oop, I should add: Please create an issue for this tidy up work, so we don't forget about it.